### PR TITLE
wicked: Add check that console usable before collecting logs

### PR DESCRIPTION
**Pull request by asmorodskyi**
Add check that console usable before collecting logs

VR: http://10.160.67.22/tests/49

### CODE REVIEW

**Comment by jlausuch** [comment]
can you point a run which shows why we need this patch? Would be nice for people to have more background about this new check.

**Comment by asmorodskyi** [comment]
> can you point a run which shows why we need this patch? Would be nice for people to have more background about this new check.

http://10.160.67.22/tests/47 - I face situation when serial_login failed . Let's skip for now **why** actually it failed but besides that I notice bad handling of this from our post_fail_hook . The tricky part that because we trying to collect as much as possible ignoring all failures around . Generally this make perfect sense that in this specific scenario it is pure waste of time - trying to type commands into login prompt. This is how I see we can workaround this without breaking general idea of "collect as much as possible no matter what"

**Comment by jlausuch** [comment]
wouldn't this PR solve the problem already for the cases when login is not successful? https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8659

**Comment by okurz** [comment]
1. I suggest to avoid IP adresses and use DNS names for links to verification runs
2. Can you point to a ticket?
3. Please describe more details about the problem in the git commit message itself so that we have this information available in the git logs and not only on github
4. This reminds me of the known issues we already have, e.g.
 * https://progress.opensuse.org/issues/36126
 * https://progress.opensuse.org/issues/34339
 * https://progress.opensuse.org/issues/49649
 * https://progress.opensuse.org/issues/31132
 * https://progress.opensuse.org/issues/34471
 * https://progress.opensuse.org/issues/46508
 * https://progress.opensuse.org/issues/48110
 * https://progress.opensuse.org/issues/46394
 * https://progress.opensuse.org/issues/54488

**Comment by asmorodskyi** [comment]
> wouldn't this [PR](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8659) solve the problem already for the cases when login is not successful?

check verification run for #8659. You will see problem which I am trying solve **here**, serial_terminal::login() is actually capable to "detect"  failure by itself. at the end of login function some commands executed in `assert_script_run` so if there is some failure login will actually fail . This PR trying to cover what will happen next in case of wicked tests - after login failure our tests trying to execute `upload_wicked_logs()` and wasting a lot of time by waiting for results of every command executed there ....

**Comment by asmorodskyi** [comment]
>     1. I suggest to avoid IP adresses and use DNS names for links to verification runs

https://infra.nue.suse.com/SelfService/Display.html?id=147092 - I am asking for DNS name for 2 weeks already ...

> 
>     2. Can you point to a ticket?

don't see any sense in creating a ticket for one line of the code .  @jlausuch  - do you think we need to create extra ticket here ?
> 
>     3. Please describe more details about the problem in the git commit message itself so that we have this information available in the git logs and not only on github

Done .

> 
>     4. This reminds me of the known issues we already have, e.g.
> 
Not sure how this could help me 

> 
>     * https://progress.opensuse.org/issues/36126
> 
>     * https://progress.opensuse.org/issues/34339
> 
>     * https://progress.opensuse.org/issues/49649
> 
>     * https://progress.opensuse.org/issues/31132
> 
>     * https://progress.opensuse.org/issues/34471
> 
>     * https://progress.opensuse.org/issues/46508
> 
>     * https://progress.opensuse.org/issues/48110
> 
>     * https://progress.opensuse.org/issues/46394
> 
>     * https://progress.opensuse.org/issues/54488

**Comment by jlausuch** [comment]
> don't see any sense in creating a ticket for one line of the code . @jlausuch - do you think we need to create extra ticket here ?
> 

For such small quick fixes, if there is no ticket reported previously, it's not necessary in my opinion.

**Comment by asmorodskyi** [comment]
> Fine with me.
> Maybe we could just call `true` or something else more mean full (e.g. `echo "CHECK CONSOLE"`), so the serial.log is more readable. But this is just a nit hint.

Done , doing new VR now

**Comment by asmorodskyi** [comment]
new VR : http://autobot/tests/59/file/serial_terminal.txt

File: lib/wickedbase.pm [hunk 1]
```diff
@@ -378,6 +378,9 @@ sub upload_wicked_logs {
     my ($self, $prefix) = @_;
     my $dir_name = $self->{name} . '_' . $prefix;
     my $logs_dir = "/tmp/$dir_name";
+    # because all later commands ignoring any error we need to prove
+    # that there is sense to do something at all
+    assert_script_run('echo "CHECK CONSOLE"', fail_message => 'Console not usable failed to collect logs');
```
Comment (jlausuch) [inline lib/wickedbase.pm]:
```suggestion
    assert_script_run('echo "CHECK CONSOLE"', fail_message => 'Console not usable. Failed to collect logs');
```

**Comment by okurz** [comment]
> > ```
> > 1. I suggest to avoid IP adresses and use DNS names for links to verification runs
> > ```
> 
> https://infra.nue.suse.com/SelfService/Display.html?id=147092 - I am asking for DNS name for 2 weeks already ...

```
$ host 10.160.67.22
22.67.160.10.in-addr.arpa domain name pointer g22.suse.de.
```

you have a DNS name already :)

> > ```
> > 2. Can you point to a ticket?
> > ```
> 
> don't see any sense in creating a ticket for one line of the code . @jlausuch - do you think we need to create extra ticket here ?

Of course you don't need to create a ticket if there isn't one already. I don't see the value. I was just asking if you *can* point to a ticket assuming there could be one already :)

**Comment by asmorodskyi** [comment]
> > > ```
> > > 1. I suggest to avoid IP adresses and use DNS names for links to verification runs
> > > ```
> > 
> > 
> > https://infra.nue.suse.com/SelfService/Display.html?id=147092 - I am asking for DNS name for 2 weeks already ...
> 
> ```
> $ host 10.160.67.22
> 22.67.160.10.in-addr.arpa domain name pointer g22.suse.de.
> ```
> 
> you have a DNS name already :)

wow  , awesome ! thanks ! Was not aware that it works like this :)


> 
> > > ```
> > > 2. Can you point to a ticket?
> > > ```
> > 
> > 
> > don't see any sense in creating a ticket for one line of the code . @jlausuch - do you think we need to create extra ticket here ?
> 
> Of course you don't need to create a ticket if there isn't one already. I don't see the value. I was just asking if you _can_ point to a ticket assuming there could be one already :)

ah I see , so I got your question wrong

**Pull request merged**